src/dbnode/persist/schema/index_entry.go

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package schema
+
+import (
+	"hash"
+)
+
+// Hash computes a hash value for this index entry using the given hash method,
+// using its ID, tags, and the computed data checksum.
+func (e *IndexEntry) Hash(h hash.Hash32) int64 {
+	h.Sum(e.ID)
+	h.Sum(e.EncodedTags)
+	// TODO: investigate performance of this; also may be neecessary to add
+	// other fields to this hash. Also subrtacting a computed value from the hash
+	// may lead to more collision prone values.
+	return int64(h.Sum32()) - e.DataChecksum
```

Comment:
Why subtract the `DataChecksum` from a hash that appears to only contain the ID and EncodedTags? Entirely plausible I'm missing something

Reply:
We need the value to identify the series and the underlying data, since that's required to perform comparisons to detect mismatches (we're looking to identify equal IDs but unequal data specifically); perhaps this should just be `hash(ID) - e.DataChecksum`